python/tskit/trees.py

```diff
@@ -3438,20 +3438,39 @@ def parse_windows(self, windows):
                                  "the only allowed strings are 'sites' or 'trees'")
         return np.array(windows)
 
-    ############################################
-    # Statistics definitions
-    ############################################
+    def __run_windowed_stat(self, windows, method, *args, **kwargs):
+        strip_dim = windows is None
+        windows = self.parse_windows(windows)
+        stat = method(*args, **kwargs, windows=windows)
+        if strip_dim:
+            stat = stat[0]
+        return stat
+
+    def __one_way_sample_set_stat(
+            self, ll_method, sample_sets, windows=None, mode=None, span_normalise=True,
+            polarised=False):
+        if sample_sets is None:
+            sample_sets = self.samples()
+        # First try to convert to a 1D numpy array. If it is, then we strip off
+        # the corresponding dimension from the output.
+        drop_dimension = False
+        sample_sets = np.array(sample_sets)
+        if len(sample_sets.shape) == 1:
```

Comment:
This won't do what you want, here: `len(np.array([[0,1,2], [3,4]]).shape)` is 1. But I get the idea.

Reply:
If we did more serious parsing of `sample_sets` we could throw a more infromative error than the somewhat inscrutible error we currently get when passing in `sample_sets = [0,1,2]`, which is I think "integers have no len()".

Reply:
It would be nice to improve this all right.

Reply:
Hmm, this is going to be hard to do well. Kicking down the road.

Reply:
See #322